akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -30,6 +30,14 @@ sealed abstract case class Uri(scheme: String, authority: Authority, path: Path,
   def isEmpty: Boolean
 
   /**
+   * Parses the given String into a Query instance.
+   */
+  def query(charset: Charset = UTF8, mode: Uri.ParsingMode = Uri.ParsingMode.Relaxed): Query = queryString match {
```

Comment:
I would suggest another approach: with deferred query parsing we never loose any information any more between the request parser and later stages. Therefore, couldn't we have a standard `lazy val query: Query` which uses parsing mode strict and UTF8 charset by default?

We could then even get completely rid of the parsing mode and just introduce another method which allows to reparse the query string with any other (like `Relaxed`) rules.